feat: Add logo customization for LiteLLM admin UI

colesmcintosh opened the pull request:
## Title

Add logo customization for LiteLLM admin UI

## Relevant issues

N/A - Feature enhancement

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [ ] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

🆕 New Feature

## Changes

### Summary
Adds a simple logo customization feature to the LiteLLM admin UI, allowing users to set a custom logo URL or use the default LiteLLM logo.

### Demo
Watch the demo here: https://www.loom.com/share/bca4e95397554c26bc7026073add115e?sid=807d547a-4c7c-4323-ae2d-91a1136d09ca

### Implementation Details
- Added logo customization page under Settings > UI Theme
- Simple UI with just logo URL input field
- Instant preview of custom logo
- Clean "Reset to Default" button that properly restores the LiteLLM logo
- Minimal changes to existing codebase

### Files Changed
- `ui/litellm-dashboard/src/components/ui_theme_settings.tsx` - New component for logo settings
- `ui/litellm-dashboard/src/contexts/ThemeContext.tsx` - New context for managing logo state
- `ui/litellm-dashboard/src/components/navbar.tsx` - Updated to use logo from context
- `ui/litellm-dashboard/src/app/page.tsx` - Added ThemeProvider and route
- `ui/litellm-dashboard/src/components/leftnav.tsx` - Added menu item
- `litellm/proxy/ui_crud_endpoints/proxy_setting_endpoints.py` - Updated to handle logo URL only
- `litellm/proxy/proxy_server.py` - Already had the get_logo_url endpoint

### Features
- Simple, focused UI with just logo URL input
- Instant preview of logo changes
- Proper handling of empty/null values to show default logo
- Logo persists across page refreshes
- Clean error handling for invalid URLs

colesmcintosh committed:
feat: Add logo customization for LiteLLM admin UI

- Add simple logo customization page under Settings > UI Theme
- Support custom logo URL with instant preview
- Clean reset to default functionality
- Wrap app with ThemeProvider for proper context
- Backend properly handles null/empty logo URLs

colesmcintosh committed:
style: Update save button to use indigo color and 'Save Changes' text for consistency

krrishdholakia commented:
is there a user issue around this? @colesmcintosh

ishaan-jaff commented on litellm/proxy/proxy_server.py:
```diff
@@ -3537,8 +3537,14 @@ async def initialize_scheduled_background_jobs(
                 )
                 await proxy_logging_obj.slack_alerting_instance.send_fallback_stats_from_prometheus()
-        
-        await cls._initialize_spend_tracking_background_jobs(scheduler=scheduler)
```
please don't change this

ishaan-jaff commented on litellm/proxy/proxy_server.py:
```diff
@@ -3585,39 +3591,6 @@ async def initialize_scheduled_background_jobs(
 
         scheduler.start()
 
-    @classmethod
```
please dont' change this

ishaan-jaff reviewed (changes requested):
LGTM, minor changes

colesmcintosh committed:
fix: Restore CloudZero background job initialization

Address PR review feedback by restoring the _initialize_spend_tracking_background_jobs method
that handles both CloudZero and Prometheus background job initialization. This ensures
critical spend tracking functionality is not lost while preserving the logo customization feature.

Changes:
- Restore _initialize_spend_tracking_background_jobs method
- Replace inline Prometheus initialization with proper method call
- Maintain CloudZero background job setup for cost tracking

ishaan-jaff reviewed (approved):
LGTM
